lib/models/LifecycleConfiguration.ts

```diff
@@ -501,15 +501,27 @@ export default class LifecycleConfiguration {
     _checkDate(date: string) {
         const isoRegex = new RegExp('^(-?(?:[1-9][0-9]*)?[0-9]{4})-' +
             '(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9])' +
-            ':([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z)?$');
+            ':([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z|[+-][0-5][0-9]:[0-5][0-9])?$');
```

Comment:
I may be nitpicking, I think the hours in the timezone cannot go beyond -12/12, so the first digit needs to be 0 or 1. (I would be fine with `[0-9]` instead but I think `[0-5]` looks particular and only seems relevant to count minutes)

```suggestion
            ':([0-5][0-9]):([0-5][0-9])(\\.[0-9]+)?(Z|[+-][0-1][0-9]:[0-5][0-9])?$');
```

Reply:
Agreed that it looks confusing. The range is actually -14/14 for some reason, If [0-1][0-9] is fine with you, we'll leave it at that. The thing is once we have checked that the format is indeed ISO-8601, further check will be done when parsing the date to ensure that the date is legal. For example 2024-02-31T00:00:00 is legal under this regex, we're not checking everything.